apps/gp2-frontend/src/events/api.ts

```diff
@@ -37,6 +38,43 @@ export const getEvents = async (
   return resp.json();
 };
 
+export const getAllFilters = ({
+  before,
+  after,
+  constraint,
+  eventType,
+}: {
+  before?: string;
+  after?: string;
+  constraint?: gp2.EventConstraint;
+  eventType?: gp2.EventType[];
+}) => {
+  const typeFilters = eventType
+    ?.map((filter) => `_tags:"${filter}"`)
+    .join(' OR ');
+  const eventFilters = getEventFilters({ before, after }, constraint);
+  return [eventFilters, typeFilters].filter(Boolean).join(' AND ');
+};
+
+export type EventListOptions = GetEventListOptions<gp2.EventConstraint> &
+  gp2.FetchEventSearchFilter;
+export const getAlgoliaEvents = (
+  client: AlgoliaClient<'gp2'>,
+  options: EventListOptions,
+) => {
+  const filters = getAllFilters(options);
+
+  return client
+    .search(['event'], options.searchQuery, {
+      filters,
+      page: options.currentPage ?? 0,
+      hitsPerPage: options.pageSize ?? 10,
+    })
+    .catch((error: Error) => {
+      throw new Error(`Could not search:${error.message}`);
+    });
+};
```

Comment:
try {
  ...
} catch

I'm not sure why we are catching the error and then throwing another one.